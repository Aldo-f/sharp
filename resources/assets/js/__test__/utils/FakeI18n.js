export default {
    install() {
        window.i18n = {};
    }
}