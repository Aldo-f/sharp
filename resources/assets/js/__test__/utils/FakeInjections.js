import Vue from 'vue';

const injectedComponents = ['$field', '$form'];

export default {
    provide() {
        return injectedComponents.reduce((injections, prop) => {
            injections[prop] = new Vue();
            return injections;
        }, {});
    }
}