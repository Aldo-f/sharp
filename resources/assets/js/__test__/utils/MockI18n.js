import Localization ,* as localizationFn from '../../mixins/Localization';

const mockLangImplementation = localeKey => `{{ ${localeKey} }}`;

export default {
    install() {
        window.i18n = {};
    },

    mockLangFunction() {
        localizationFn.lang = jest.fn(mockLangImplementation);
        Localization.methods.l = jest.fn(mockLangImplementation);
    }
}