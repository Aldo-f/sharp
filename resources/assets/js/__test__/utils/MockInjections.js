import Vue from 'vue';

const injectedComponents = {
    $field:{},
    $form:{
        data: () => ({
            errors: {}
        })
    },
    $tab:{}
};

export default {
    provide() {
        return Object.keys(injectedComponents).reduce((injections, compName) => {
            let Comp = Vue.extend(injectedComponents[compName]);
            injections[compName] = new Comp();
            return injections;
        }, {});
    }
}