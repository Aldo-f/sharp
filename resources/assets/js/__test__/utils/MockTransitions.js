export default {
    install() {
        window.getComputedStyle = jest.fn(() => ({
            animationDelay:'',
            animationDuration:'',
            transitionDelay:'',
            transitionDuration:''
        }));
    }
}