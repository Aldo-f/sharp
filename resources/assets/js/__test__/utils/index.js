export { default as MockInjections } from './MockInjections';
export { default as MockI18n } from './MockI18n';
export { default as QueryComponent } from './QueryComponent';
export { default as MockTransitions } from './MockTransitions';
export { mockSFC } from './mockSFC';

export function wait(delay) {
    return new Promise(resolve => {
        setTimeout(resolve, delay);
    });
}