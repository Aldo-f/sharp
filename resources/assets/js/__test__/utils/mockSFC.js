export function mockSFC (sfcModule, { template, render, ...extend }={}, mockOptions={}) {

    let { isJsFile } = mockOptions;

    if(isJsFile) {
        sfcModule.__oldDefault__ = { ...sfcModule.default }
    }

    Object.assign(sfcModule.default, {
        mixins: [],
        'extends': extend,
        template: template,
        render: !template ? render || function() { return this._v(`__MOCKED_SFC_${sfcModule.name}__\n`) } : null,
        watch: {},
        beforeCreate: ()=>{},
        created: ()=>{},
        beforeMount: ()=>{},
        mounted: ()=>{},
        beforeUpdate: ()=>{},
        updated: ()=>{},
        beforeDestroy: ()=>{},
        destroyed: ()=>{},
        data: ()=>({}),
        provide: {},
        inject: [],
    });

    //console.log(sfcModule);
    return sfcModule.default;
}

export function unmockSFC(sfcModule) {
    if(sfcModule.__oldDefault__) {
        sfcModule.default = sfcModule.__oldDefault__;
        return;
    }

    let { 'default':defaultObj, ...options } = sfcModule;
    sfcModule.default = options;
}