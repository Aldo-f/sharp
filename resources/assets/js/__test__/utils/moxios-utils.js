import moxios from 'moxios';

export function nextRequestFulfilled(config) {
    return new Promise(resolve => {
        moxios.wait(async () => {
            let request = moxios.requests.mostRecent();
            let response = await request.respondWith(config);
            resolve(response); // moxios Response object
        })
    });
}
