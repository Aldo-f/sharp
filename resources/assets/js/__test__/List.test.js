import Vue from 'vue/dist/vue.common';
import List from '../components/form/fields/list/List.vue';
import FieldDisplay from '../components/form/FieldDisplay';

import { MockInjections, MockTransitions } from './utils';

describe('list-field', () => {
    Vue.component('sharp-list', List);
    Vue.component('sharp-field-display', FieldDisplay);

    Vue.use(MockTransitions);

    beforeEach(()=>{
        document.body.innerHTML = `    
            <div id="app">
                <sharp-list :value="value" 
                            :field-layout="{ 
                                item:[
                                    [ {key:'name'} ]
                                ]
                            }"
                            :item-fields="{ name: { type:'text' } }"
                            :addable="addable" 
                            :sortable="sortable"
                            :removable="removable"
                            :collapsed-item-template="'<span> {{name}} </span>'"
                            :max-item-count="5"
                            item-id-attribute="id"
                            :read-only="readOnly"
                            @input="inputEmitted">
                </sharp-list>
            </div>
        `
    });

    it('can mount empty list field', async () => {
        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('can mount filled list field', async () => {
        await createVm({
            data:()=>({
                value: [{id:0, name:'Antoine'}]
            })
        });

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('can mount collapsed list field', async () => {
        let $list = await createVm({
            data:()=>({
                value: [{id:0, name:'Antoine'}]
            }),
        });

        $list.dragActive = true;

        await Vue.nextTick();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('emit input on init to have list and value equals by reference (sync changes)', async () => {
        let inputEmitted = jest.fn();

        let $list = await createVm({
            methods: {
                inputEmitted
            }
        });


        expect(inputEmitted).toHaveBeenCalledTimes(1);
        expect(inputEmitted).toHaveBeenCalledWith($list.list);
    });
});

async function createVm(customOptions={}) {

    const vm = new Vue({
        el: '#app',
        mixins: [MockInjections, customOptions],

        props:['readOnly', 'addable', 'sortable', 'removable'],

        'extends': {
            data:() => ({
                value: null
            }),
            methods: {
                inputEmitted: ()=>{}
            }
        }
    });

    await Vue.nextTick();

    return vm.$children[0];
}