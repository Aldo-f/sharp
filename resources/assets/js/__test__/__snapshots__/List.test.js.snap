// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list-field can mount collapsed list field 1`] = `
<div id="app">
    <div class="SharpList">
        <!---->
        <div>
            <div>
                <div class="SharpList__item SharpList__item--collapsed">
                    <div class="SharpModule__inner">
                        <div class="SharpModule__content">
                            <div><span> Antoine </span></div>
                        </div>
                    </div>
                    <!---->
                </div>
            </div>
            <!---->
        </div>
        <!---->
    </div>
</div>
`;

exports[`list-field can mount empty list field 1`] = `
<div id="app">
    <div class="SharpList">
        <!---->
        <div>
            <div></div><button type="button" class="SharpButton SharpButton--secondary SharpList__add-button">Ajouter un élément</button></div>
        <!---->
    </div>
</div>
`;

exports[`list-field can mount filled list field 1`] = `
<div id="app">
    <div class="SharpList">
        <!---->
        <div>
            <div>
                <div class="SharpList__item">
                    <div class="SharpModule__inner">
                        <div class="SharpModule__content">
                            <div class="SharpGrid">
                                <div class="row">
                                    <div class="col-md">
                                        <div class="SharpFieldContainer SharpForm__form-item"><label class="SharpForm__label" style="display: none;">
         <!----></label>
                                            <!----><input type="text" class="SharpText">
                                            <div class="SharpForm__form-requirement"></div> <small class="SharpForm__help-message"></small></div>
                                    </div>
                                </div>
                            </div>
                            <!---->
                        </div>
                    </div>
                    <!---->
                </div>
            </div><button type="button" class="SharpButton SharpButton--secondary SharpList__add-button">Ajouter un élément</button></div>
        <!---->
    </div>
</div>
`;

exports[`list-field can mount non addable, non removable, non sortable list field 1`] = `
<div id="app">
    <div class="SharpList">
        <!---->
        <div>
            <div>
                <div class="SharpList__item">
                    <div class="SharpModule__inner">
                        <div class="SharpModule__content">
                            <div class="SharpGrid">
                                <div class="row">
                                    <div class="col-md">
                                        <div class="SharpFieldContainer SharpForm__form-item"><label class="SharpForm__label" style="display: none;">
         <!----></label>
                                            <!----><input type="text" class="SharpText">
                                            <div class="SharpForm__form-requirement"></div> <small class="SharpForm__help-message"></small></div>
                                    </div>
                                </div>
                            </div>
                            <!---->
                        </div>
                    </div>
                    <!---->
                </div>
            </div>
            <!---->
        </div>
        <!---->
    </div>
</div>
`;

exports[`list-field can mount with full list 1`] = `
<div id="app">
    <div class="SharpList">
        <!---->
        <div>
            <div>
                <div class="SharpList__item">
                    <div class="SharpModule__inner">
                        <div class="SharpModule__content">
                            <div class="SharpGrid">
                                <div class="row">
                                    <div class="col-md">
                                        <div class="SharpFieldContainer SharpForm__form-item"><label class="SharpForm__label" style="display: none;">
         <!----></label>
                                            <!----><input type="text" class="SharpText">
                                            <div class="SharpForm__form-requirement"></div> <small class="SharpForm__help-message"></small></div>
                                    </div>
                                </div>
                            </div>
                            <!---->
                        </div>
                    </div>
                    <!---->
                </div>
                <div class="SharpList__item">
                    <div class="SharpModule__inner">
                        <div class="SharpModule__content">
                            <div class="SharpGrid">
                                <div class="row">
                                    <div class="col-md">
                                        <div class="SharpFieldContainer SharpForm__form-item"><label class="SharpForm__label" style="display: none;">
         <!----></label>
                                            <!----><input type="text" class="SharpText">
                                            <div class="SharpForm__form-requirement"></div> <small class="SharpForm__help-message"></small></div>
                                    </div>
                                </div>
                            </div>
                            <!---->
                        </div>
                    </div>
                    <!---->
                </div>
                <div class="SharpList__item">
                    <div class="SharpModule__inner">
                        <div class="SharpModule__content">
                            <div class="SharpGrid">
                                <div class="row">
                                    <div class="col-md">
                                        <div class="SharpFieldContainer SharpForm__form-item"><label class="SharpForm__label" style="display: none;">
         <!----></label>
                                            <!----><input type="text" class="SharpText">
                                            <div class="SharpForm__form-requirement"></div> <small class="SharpForm__help-message"></small></div>
                                    </div>
                                </div>
                            </div>
                            <!---->
                        </div>
                    </div>
                    <!---->
                </div>
                <div class="SharpList__item">
                    <div class="SharpModule__inner">
                        <div class="SharpModule__content">
                            <div class="SharpGrid">
                                <div class="row">
                                    <div class="col-md">
                                        <div class="SharpFieldContainer SharpForm__form-item"><label class="SharpForm__label" style="display: none;">
         <!----></label>
                                            <!----><input type="text" class="SharpText">
                                            <div class="SharpForm__form-requirement"></div> <small class="SharpForm__help-message"></small></div>
                                    </div>
                                </div>
                            </div>
                            <!---->
                        </div>
                    </div>
                    <!---->
                </div>
                <div class="SharpList__item">
                    <div class="SharpModule__inner">
                        <div class="SharpModule__content">
                            <div class="SharpGrid">
                                <div class="row">
                                    <div class="col-md">
                                        <div class="SharpFieldContainer SharpForm__form-item"><label class="SharpForm__label" style="display: none;">
         <!----></label>
                                            <!----><input type="text" class="SharpText">
                                            <div class="SharpForm__form-requirement"></div> <small class="SharpForm__help-message"></small></div>
                                    </div>
                                </div>
                            </div>
                            <!---->
                        </div>
                    </div>
                    <!---->
                </div>
            </div>
            <!---->
        </div>
        <!---->
    </div>
</div>
`;
